src/avatar_action.cpp

```diff
@@ -1000,20 +1001,25 @@ void avatar_action::use_item( avatar &you, item_location &loc )
             return;
         }
 
-        const item &it = *loc.get_item();
-        if( it.has_flag( "ALLOWS_REMOTE_USE" ) ) {
-            use_loc = true;
+        if( loc->has_flag( "ALLOWS_REMOTE_USE" ) ) {
+            use_in_place = true;
         } else {
-            int obtain_cost = loc.obtain_cost( you );
-            loc.obtain( you );
-            // This method only handles items in te inventory, so refund the obtain cost.
+            const int obtain_cost = loc.obtain_cost( you );
+            loc = item_location( you, &you.i_at( loc.obtain( you ) ) );
+            if( !loc ) {
```

Comment:
I believe this will never trigger. `you.i_at` returns a reference, therefor the pointer passed to `item_location` will always be non-null, which in turn makes `loc` always valid (even through it refers to the null-item returned by `i_at`).

You need to check either the result of `i_at` (via `item::is_null`) or check the result of `loc.obtain`.

Reply:
Yes, you're right. Good catch